crates/nu-parser/src/parser.rs

```diff
@@ -449,24 +449,20 @@ fn parse_short_flags(
 ) -> Option<Vec<Flag>> {
     let arg_span = spans[*spans_idx];
 
-    let arg_contents = working_set.get_span_contents(arg_span);
+    let arg_contents = working_set.get_span_contents(arg_span).to_vec();
 
     if arg_contents.starts_with(b"-") && arg_contents.len() > 1 {
         let short_flags = &arg_contents[1..];
         let mut found_short_flags = vec![];
         let mut unmatched_short_flags = vec![];
-        for short_flag in short_flags.iter().enumerate() {
-            let short_flag_char = char::from(*short_flag.1);
+        for (i, short_flag) in short_flags.iter().enumerate() {
+            let short_flag_char = char::from(*short_flag);
             let orig = arg_span;
-            let short_flag_span = Span::new(
-                orig.start + 1 + short_flag.0,
-                orig.start + 1 + short_flag.0 + 1,
-            );
+            let short_flag_span = Span::new(orig.start + 1 + i, orig.start + 1 + i + 1);
```

Comment:
This line previously already made the assumption that a short flag is only an ASCII character and requires just one byte. Our parsing of definitions apparently doesn't check for that causing downstream panics. I opened an issue for that (#8821)

Reply:
Ohh good catch!

Reply:
@sholderbach would you like #8821 to get fixed first, or are you okay with merging this before?

Reply:
nvm, it seems like #8849 fixed the issue!